homeassistant/components/modbus/bit_sensor.py

```diff
@@ -0,0 +1,132 @@
+"""Support for Modbus Coil and Discrete Input sensors."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+from pymodbus.exceptions import ConnectionException, ModbusException
+from pymodbus.pdu import ExceptionResponse
+
+from homeassistant.components.binary_sensor import BinarySensorEntity
+from homeassistant.const import (
+    CONF_ADDRESS,
+    CONF_DEVICE_CLASS,
+    CONF_NAME,
+    CONF_SCAN_INTERVAL,
+    CONF_SLAVE,
+)
+from homeassistant.helpers.event import async_track_time_interval
+from homeassistant.helpers.typing import (
+    ConfigType,
+    DiscoveryInfoType,
+    HomeAssistantType,
+)
+
+from .const import CALL_TYPE_COIL, CONF_INPUT_TYPE, MODBUS_DOMAIN
+from .modbus import ModbusHub
+
+CONF_BIT_SENSORS = "bit_sensors"
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(
```

Comment:
This will not be called, unless you add BIT_SENSOR/BIT_SENSORS to the loop in modbus.py (look for e.g. CLIMATE)